src/base/BaseStrategy.sol

```diff
@@ -0,0 +1,396 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+import {IERC20, Math, SafeERC20} from "src/Common.sol";
+import {Vault} from "src/Vault.sol";
+
+/**
+ * @title BaseStrategy
+ * @author Yieldnest
+ * @notice This contract is a base strategy for any underlying protocol.
+ * vault.
+ */
+abstract contract BaseStrategy is Vault {
+    /// @notice Role for allocator permissions
+    bytes32 public constant ALLOCATOR_ROLE = keccak256("ALLOCATOR_ROLE");
+
```

Comment:
how about we add a STRATEGY_VERSION = "0.1.0"

like here https://github.com/yearn/tokenized-strategy/blob/master/src/TokenizedStrategy.sol#L346